### PR TITLE
#8543 Make tutorial compatible with being module plugin.

### PR DESCRIPTION
**Pull request by alexander-fedorenko**
## Description
There is a problem caused by the fact that plugins reducers are loaded after LOCATION_CHANGE action dispatched. It makes plugin epic first run skipped. This PR corrects this behavior

- Improved REDUCERS_LOADED action to make it pass list of added reducers as a payload.
- Updated switchTutorialEpic to make it actually triggered once tutorial reducer is added & one of the subsequent actions dispatched.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**
#8543 

**What is the new behavior?**
Epic is not skipped, tracked actions list is extended to work correctly both on initial load & on location changes.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/utils/StateUtils.js [hunk 1]
```diff
@@ -15,6 +15,18 @@ import isEmpty from 'lodash/isEmpty';
 import {BehaviorSubject, Subject} from 'rxjs';
 import {normalizeName} from "./PluginsUtils";
 
+export const REDUCERS_LOADED = 'REDUCERS_LOADED';
+
+/**
+ * Action that should be dispatched whenever new reducers are added
+ * @param reducers
```
Comment (MV88) [inline web/client/utils/StateUtils.js:22]:
```suggestion
 * @param {string[]} reducers list of reducers loaded
```

**Review by MV88 — CHANGES_REQUESTED** [review]
- [ ] I noticed then you use tutorial in map and then you open either a new or old dashboard the tutorial there it does not start
this was happening on last release and on QA

https://user-images.githubusercontent.com/11991428/188470783-b0f13da6-d4df-4ab6-a1ed-0277b796aeea.mp4


there is also something weird, it seems it is continuing the map tutorial
if you instead go directly to a dashboard it starts the correct tutorial

this is not affecting story

@tdipisa do we need to pen a dedicated issue for this?

**Comment by tdipisa** [comment]
>@tdipisa do we need to pen a dedicated issue for this?

The problem seems  to be effectively connected to this bug and therefore it should be solved as part of this PR. Can you please check @alexander-fedorenko?

**Comment by alexander-fedorenko** [comment]
@MV88 Thanks for review! All listed requests for changes are applied.

Issue with incorrect behavior of Tutorial was related to the `LOCATION_CHANGE` event and to the fact that outer stream in `switchTutorialEpic` was muted at some pages upon movement from one route with tutorial plugin active to another. As a result in some cases `LOCATION_CHANGE` was intercepted by epic on one route, while inner stream and one of the intercepted actions ran on another page. Inner stream was relying on information obtained from outer action and route information was incorrect. Changing logic to use state value of router instead solved the issue.

I cross-checked code of another plugins where epics intercepting `LOCATION_CHANGE` and didn't see any other place where similar situation can appear. Issue can only appear if epic returns nested stream and rely on information from outer action.

**Review by MV88 — APPROVED** [review]
LGTM

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Comment by ElenaGallo** [comment]
Test passed @alexander-fedorenko please backport to 2022.02.xx. Thanks

**Pull request merged**